scormxblock/static/js/src/scormxblock.js

```diff
@@ -4,105 +4,165 @@ function ScormXBlock(runtime, element, settings) {
 
     this.LMSInitialize = function(){
       console.log('LMSInitialize');
-      return "true";
+      lastError = 0;
+      return 'true';
     };
 
     this.LMSFinish = function() {
-      console.log("LMSFinish");
-      return "true";
+      console.log('LMSFinish');
+      return StoreValues();
     };
 
     this.LMSGetValue = GetValue;
     this.LMSSetValue = SetValue;
 
     this.LMSCommit = function() {
-        console.log("LMSCommit");
-        return "true";
+      console.log('LMSCommit');
+      return StoreValues();
     };
 
     this.LMSGetLastError = function() {
-      console.log("GetLastError");
-      return "0";
+      console.log('GetLastError');
+      return lastError;
     };
 
     this.LMSGetErrorString = function(errorCode) {
-      console.log("LMSGetErrorString");
-      return "Some Error";
+      console.log('LMSGetErrorString');
+      return SCORM_12_ERRORS[errorCode.toString()];
     };
 
     this.LMSGetDiagnostic = function(errorCode) {
-      console.log("LMSGetDiagnostic");
-      return "Some Diagnostice";
+      console.log('LMSGetDiagnostic');
+      return 'Some Diagnostice';
     }
   }
 
   function SCORM_2004_API(){
     this.Initialize = function(){
       console.log('LMSInitialize');
-      return "true";
+      return 'true';
     };
 
     this.Terminate = function() {
-      console.log("LMSFinish");
-      return "true";
+      console.log('LMSFinish');
+      return StoreValues();
     };
 
     this.GetValue = GetValue;
     this.SetValue = SetValue;
 
     this.Commit = function() {
-        console.log("LMSCommit");
-        return "true";
+      console.log('LMSCommit')
+      return StoreValues();
     };
 
     this.GetLastError = function() {
-      console.log("GetLastError");
-      return "0";
+      console.log('GetLastError');
+      return '0';
     };
 
     this.GetErrorString = function(errorCode) {
-      console.log("LMSGetErrorString");
-      return "Some Error";
+      console.log('LMSGetErrorString');
+      return 'Some Error';
     };
 
     this.GetDiagnostic = function(errorCode) {
-      console.log("LMSGetDiagnostic");
-      return "Some Diagnostice";
+      console.log('LMSGetDiagnostic');
+      return 'Some Diagnostice';
     }
   }
 
+  var lastError = 301;
+
+  var SCORM_12_ERRORS = {
```

Comment:
Could you please add a inline comment about where these errors came from? @andrey-canon Some documentation, or link.